src/Targets/Microsoft.Managed.DesignTime.targets

```diff
@@ -25,6 +25,11 @@
     <SuppressOutOfDateMessageOnBuild Condition="'$(SuppressOutOfDateMessageOnBuild)' == ''">true</SuppressOutOfDateMessageOnBuild>
   </PropertyGroup>
 
+  <!-- If the project we're compiling has a ruleset, make sure we check it for up-to-date checks. -->
+  <ItemGroup Condition="'$(CodeAnalysisRuleSet)' != ''">
```

Comment:
Make sure to verify that this doesn't do anything wonky in the old PS (the designtime targets are imported by non-SDK projects as well).